resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -162,6 +161,8 @@ private[spark] class Client(
                   .endEnv()
                 .addToEnv(sslEnvs: _*)
                 .withPorts(containerPorts.asJava)
+                .withNewReadinessProbe().withHttpGet(probePingHttpGet).endReadinessProbe()
+                .withNewLivenessProbe().withHttpGet(probePingHttpGet).endLivenessProbe()
```

Comment:
The liveness could be useful to restart our server if it fails for some reason, but we would need an initial delay seconds, since otherwise, we'll have it fail early on, before the server starts listening.